sktime/forecasting/reconcile.py

```diff
@@ -0,0 +1,355 @@
+# -*- coding: utf-8 -*-
+# !/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements reconciled forecasters for hierarchical data."""
+
+__all__ = ["ReconcilerForecaster"]
+__author__ = [
+    "ciaran-g",
+]
+
+# todo: include the reconciler transformers?
+# todo: top down historical proportions?
+
+from warnings import warn
+
+import numpy as np
+import pandas as pd
+from numpy.linalg import inv
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+from sktime.transformations.hierarchical.aggregate import _check_index_no_total
+from sktime.transformations.hierarchical.reconcile import (
```

Comment:
hm... should this move somewhere? The "parent child", or more generally, graph stuff, could fit in `datatypes` as utilities for the hierarchical data. Although, for the moment, let´s leave it where it is to avoid refactor complexity sneaking in. Worth thinking about though.